new_dawn_server/actions/api/tests/test_like.py

```diff
@@ -0,0 +1,97 @@
+from new_dawn_server.actions.constants import ActionType, EntityType
+from new_dawn_server.actions.models import UserAction
+from new_dawn_server.users.models import Account
+from tastypie.test import ResourceTestCaseMixin
+
+
+class UserActionTest(ResourceTestCaseMixin, TestCase):
+    def setUp(self):
+        super().setUp()
+        self.user1_aruguments = {
+            "first_name": "test",
+            "last_name": "user",
+            "username": "duck",
+            "password": "test-pwd",
+            "birthday": "1990-01-01",
+            "phone_number": "+12345678900",
+            "gender": "M",
+            "description": "nice",
+            "employer": "MANMAN",
+            "hometown": "NY",
+            "job_title": "CEO",
+            "profile_photo_url": "www",
+            "school": "NYU",
+            "smoke": "Socially",
+            "city_preference":
+                [
+                    {
+                        "city": "New York",
+                        "country": "US",
+                        "state": "NY"
+                    },
+                    {
+                        "city": "Salt Lake City",
+                        "country": "US",
+                        "state": "UT"
+                    }
+                ]
+        }
+        self.user2_arguments = {
+            "first_name": "test2",
+            "last_name": "user2",
+            "name": "goodgirl",
+            "username": "duck2",
+            "password": "test-pwd2",
+            "birthday": "1990-01-01",
+            "phone_number": "+12345678900",
+            "gender": "M",
+            "description": "nice",
+            "employer": "MANMAN",
+            "hometown": "NY",
+            "job_title": "CEO",
+            "profile_photo_url": "www",
+            "school": "NYU",
+            "smoke": "Socially",
+            "city_preference":
+                [
+                    {
+                        "city": "New York",
+                        "country": "US",
+                        "state": "NY"
+                    },
+                    {
+                        "city": "Salt Lake City",
+                        "country": "US",
+                        "state": "UT"
+                    }
+                ]
+        }
+        self.api_client.post(
+            "/api/v1/register/", format="json", data=self.user1_aruguments)
+        self.api_client.post(
+            "/api/v1/register/", format="json", data=self.user2_arguments)
+
+        self.like_argument = {
+            "action_type": ActionType.LIKE.value,
+            "entity_id": 1,
+            "entity_type": EntityType.MAIN_IMAGE.value,
+            "user_account_from": "/api/v1/account/1/",
+            "user_account_to": "/api/v1/account/2/"
+        }
+
+    def test_like_user(self):
+        res = self.api_client.post(
+            "/api/v1/user_action/", format="json", data=self.like_argument
+        )
+        res_data = json.loads(res.content)
```

Comment:
Even though I can successfully create Account and UserAction objects, but I get error message from post method: "{'error': "The 'city_preference' field has no data and doesn't allow a null value."}".

Reply:
Let me take a look tonight.

Reply:
Can you try adding` null = True` here and try again? https://github.com/new-dawn/new_dawn_server/blob/master/new_dawn_server/users/api/resources.py#L198

Reply:
yes it works now..